### PR TITLE
Issue/ruby rails upgrade

### PR DESCRIPTION
**Pull request by bogdanadrianmarc**
This PR upgrades ruby to `3.3.5` and rails to `7.2.2`

### CODE REVIEW

File: app/assets/stylesheets/ppd.scss [hunk 1]
```diff
@@ -167,7 +168,7 @@
   }
 
   .transaction-category {
-    background-color: lighten( $lr-logo-grey, 20% );
+    background-color: lighten( #a7a5a6, 20% );
```
Comment (joescottdave) [inline app/assets/stylesheets/ppd.scss:171]: What was the issue here?

Comment (jonrandahl) [reply app/assets/stylesheets/ppd.scss:171]: The variable value was lost on import/compilation and this was felt to be the quickest way to resolve. Additional ticket on this was requested.

Comment (joescottdave) [reply app/assets/stylesheets/ppd.scss:171]: Does the other ticket exist @bogdanadrianmarc @jonrandahl ?

Comment (bogdanadrianmarc) [reply app/assets/stylesheets/ppd.scss:171]: @joescottdave Had a discussion with @jonrandahl the other day about this, I've put a summary of it in here now: https://github.com/epimorphics/lr_common_styles/issues/75

Comment (joescottdave) [reply app/assets/stylesheets/ppd.scss:171]: Great 👍

**Review by joescottdave — APPROVED** [review]
LGTM

**Pull request merged**